src/EditorFeatures/Core/Shared/Preview/PreviewSolutionCrawlerRegistrationService.cs

```diff
@@ -18,11 +20,15 @@ internal class PreviewSolutionCrawlerRegistrationService : ISolutionCrawlerRegis
         private static readonly ConditionalWeakTable<Workspace, CancellationTokenSource> s_cancellationTokens =
             new ConditionalWeakTable<Workspace, CancellationTokenSource>();
 
+        private readonly IDiagnosticService _service;
         private readonly IIncrementalAnalyzerProvider _provider;
 
         [ImportingConstructor]
-        public PreviewSolutionCrawlerRegistrationService(IDiagnosticAnalyzerService diagnosticService)
+        public PreviewSolutionCrawlerRegistrationService(IDiagnosticService service, IDiagnosticAnalyzerService diagnosticService)
```

Comment:
These names are super confusing.  IDiagnosticsService is called "service".  but "diagnosticsService" is actually a DiagnosticAnalyzerservice?